In [122]:
#import libs 

import pandas as pd

import plotly as py
import plotly.express as px
import plotly.graph_objs as go 

In [82]:
filepath = r"C:\Users\krupa\OneDrive\Documents\Krups coding\Data analysis\wiki data\Country_data.csv"

data = pd.read_csv(filepath)
data.head()
    

Country  Total in km2  Land in km2  Water in km2  Water %    HDI  \
0     Russia    17098246.0   16378410.0    16378410.0      4.2  0.822   
1     Canada     9984670.0    9093507.0     9093507.0      8.9  0.936   
2     Brazil     8510346.0    8460415.0     8460415.0      0.6  0.754   
3      India     3287263.0    2973190.0     2973190.0      9.6  0.633   
4  Argentina     2796427.0    2736690.0     2736690.0      1.6  0.842   

   %HDI Growth  IMF Forecast GDP(Nominal)  World Bank Forecast GDP(Nominal)  \
0         0.29                  1862470.0                         2240422.0   
1         0.25                  2117805.0                         2139840.0   
2         0.38                  2126809.0                         1920096.0   
3         0.88                  3732224.0                         3385090.0   
4         0.09                   621833.0                          632770.0   

   UN Forecast GDP(Nominal)  IMF Forecast GDP(PPP)  \
0                 1778782.0              5056479.0   
1                 1988336.0              2378973.0   
2                 1608981.0              4101022.0   
3                 3201471.0             13119622.0   
4                  487227.0              1239515.0   

   World Bank Forecast GDP(PPP)  CIA Forecast GDP(PPP)  Internet Users  \
0                     5326855.0              3875690.0     129800000.0   
1                     2273489.0              1742790.0      33950632.0   
2                     3837261.0              2989430.0     165300000.0   
3                    11874583.0              8443360.0     881250000.0   
4                     1225435.0               893310.0      40024673.0   

  UN Continental Region UN Statistical Subregion  Population 2022  \
0                Europe           Eastern Europe     1.447133e+08   
1              Americas         Northern America     3.845433e+07   
2              Americas            South America     2.153135e+08   
3                  Asia            Southern Asia     1.417173e+09   
4              Americas            South America     4.551032e+07   

   Population 2023  Population %Change  
0     1.444444e+08               -0.19  
1     3.878129e+07                0.85  
2     2.164224e+08                0.52  
3     1.428628e+09                0.81  
4     4.577388e+07                0.58

In [83]:
#stat info. about dataset 
data.describe()

Total in km2   Land in km2  Water in km2     Water %         HDI  \
count  1.420000e+02  1.420000e+02  1.420000e+02  142.000000  142.000000   
mean   6.516930e+05  6.304762e+05  6.304762e+05    2.538099    0.704324   
std    1.846820e+06  1.763518e+06  1.763518e+06    4.148328    0.152592   
min    2.600000e+01  2.600000e+01  2.600000e+01    0.000000    0.385000   
25%    2.669100e+04  2.549475e+04  2.549475e+04    0.000000    0.585250   
50%    1.257585e+05  1.221600e+05  1.221600e+05    1.050000    0.717500   
75%    5.907288e+05  5.784400e+05  5.784400e+05    2.975000    0.815000   
max    1.709825e+07  1.637841e+07  1.637841e+07   27.800000    0.962000   

       %HDI Growth  IMF Forecast GDP(Nominal)  \
count   140.000000               1.360000e+02   
mean      0.539571               3.268823e+05   
std       0.396370               7.636126e+05   
min       0.000000               6.300000e+01   
25%       0.230000               1.145125e+04   
50%       0.445000               3.256650e+04   
75%       0.762500               3.084302e+05   
max       1.940000               4.429838e+06   

       World Bank Forecast GDP(Nominal)  UN Forecast GDP(Nominal)  \
count                      1.370000e+02              1.380000e+02   
mean                       3.055762e+05              2.945361e+05   
std                        7.201154e+05              7.357661e+05   
min                        6.000000e+01              6.000000e+01   
25%                        1.049200e+04              6.367750e+03   
50%                        2.995700e+04              2.714700e+04   
75%                        2.641820e+05              2.211575e+05   
max                        4.231141e+06              4.940878e+06   

       IMF Forecast GDP(PPP)  World Bank Forecast GDP(PPP)  \
count           1.370000e+02                  1.410000e+02   
mean            6.627371e+05                  5.994211e+05   
std             1.569281e+06                  1.428972e+06   
min             6.300000e+01                  6.100000e+01   
25%             2.835600e+04                  1.935800e+04   
50%             7.854700e+04                  6.491000e+04   
75%             5.484650e+05                  5.063390e+05   
max             1.311962e+07                  1.187458e+07   

       CIA Forecast GDP(PPP)  Internet Users  Population 2022  \
count           1.390000e+02    1.420000e+02     1.420000e+02   
mean            4.629539e+05    2.245352e+07     3.650124e+07   
std             1.088335e+06    8.024149e+07     1.252199e+08   
min             5.000000e+01    5.520000e+03     1.131200e+04   
25%             1.830500e+04    5.799232e+05     2.096591e+06   
50%             5.692000e+04    3.012851e+06     9.488042e+06   
75%             3.717800e+05    1.232198e+07     3.236403e+07   
max             8.443360e+06    8.812500e+08     1.417173e+09   

       Population 2023  Population %Change  
count     1.420000e+02          142.000000  
mean      3.689126e+07            1.166408  
std       1.262483e+08            1.124469  
min       1.139600e+04           -3.640000  
25%       2.101970e+06            0.522500  
50%       9.507554e+06            1.055000  
75%       3.314716e+07            1.960000  
max       1.428628e+09            4.980000

In [84]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142 entries, 0 to 141
Data columns (total 19 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Country                           142 non-null    object 
 1   Total in km2                      142 non-null    float64
 2   Land in km2                       142 non-null    float64
 3   Water in km2                      142 non-null    float64
 4   Water %                           142 non-null    float64
 5   HDI                               142 non-null    float64
 6   %HDI Growth                       140 non-null    float64
 7   IMF Forecast GDP(Nominal)         136 non-null    float64
 8   World Bank Forecast GDP(Nominal)  137 non-null    float64
 9   UN Forecast GDP(Nominal)          138 non-null    float64
 10  IMF Forecast GDP(PPP)             137 non-null    float64
 11  World Bank Forecast GDP(PPP)      141 non-null    float64
 12  CIA Fore

In [85]:
data.shape

(142, 19)

The dataset consists of 19 columns and 142 rows. 16 out of the 19 columns have float data types and 3 columns have string object data types.

The dataset consists of some interesting information. Before we dive in, let's check for any inconsistencies and missing values.

In [88]:
#locate any missing vals
print('missing values (%) per column: \n', 100*data.isnull().mean())

missing values (%) per column: 
 Country                             0.000000
Total in km2                        0.000000
Land in km2                         0.000000
Water in km2                        0.000000
Water %                             0.000000
HDI                                 0.000000
%HDI Growth                         1.408451
IMF Forecast GDP(Nominal)           4.225352
World Bank Forecast GDP(Nominal)    3.521127
UN Forecast GDP(Nominal)            2.816901
IMF Forecast GDP(PPP)               3.521127
World Bank Forecast GDP(PPP)        0.704225
CIA Forecast GDP(PPP)               2.112676
Internet Users                      0.000000
UN Continental Region               0.000000
UN Statistical Subregion            0.000000
Population 2022                     0.000000
Population 2023                     0.000000
Population %Change                  0.000000
dtype: float64


Most of the missing values are within the GDP forecast columns. The column with the most missing values is the IMF Forecast GDP (Nominal). 

In [90]:
#non-numerical data columns
non_numeric_data = list(data.select_dtypes(exclude = ('int', 'float')).columns)
print(f'Columns without numeric data: {", ".join(non_numeric_data)}.')

Columns without numeric data: Country, UN Continental Region, UN Statistical Subregion.


In [91]:
#check for any duplications 
print('Duplication overall: ', 100*(len(data)-len(data.drop_duplicates()))/len(data))
print('Duplication in Country column: ', 100*(len(data['Country'])-len(data['Country'].drop_duplicates()))/len(data['Country']))

Duplication overall:  0.0
Duplication in Country column:  0.0


There are no overall duplicates or duplicates in any of the country names. We can expect there to be duplicates in the region and sub-region columns.

In [93]:
#Let's see how the data is split geographically 
continents_df = data['UN Continental Region'].value_counts().to_frame().reset_index().rename(columns = {'index':'UN Continental Region', 'UN Continental Region':'Quantity'})
continents_df

UN Continental Region  Quantity
0                Africa        42
1                  Asia        34
2              Americas        30
3                Europe        29
4               Oceania         7

In [127]:
fig = px.pie(continents_df, names = 'UN Continental Region', values = 'Quantity',
               title = 'Data split by Continents',
               )

In [128]:
fig.show()

let's take a look at the land area for each country 

In [72]:
data.head()

Country  Total in km2  Land in km2  Water in km2  Water %    HDI  \
0     Russia    17098246.0   16378410.0    16378410.0      4.2  0.822   
1     Canada     9984670.0    9093507.0     9093507.0      8.9  0.936   
2     Brazil     8510346.0    8460415.0     8460415.0      0.6  0.754   
3      India     3287263.0    2973190.0     2973190.0      9.6  0.633   
4  Argentina     2796427.0    2736690.0     2736690.0      1.6  0.842   

   %HDI Growth  IMF Forecast GDP(Nominal)  World Bank Forecast GDP(Nominal)  \
0         0.29                  1862470.0                         2240422.0   
1         0.25                  2117805.0                         2139840.0   
2         0.38                  2126809.0                         1920096.0   
3         0.88                  3732224.0                         3385090.0   
4         0.09                   621833.0                          632770.0   

   UN Forecast GDP(Nominal)  IMF Forecast GDP(PPP)  \
0                 1778782.0              5056479.0   
1                 1988336.0              2378973.0   
2                 1608981.0              4101022.0   
3                 3201471.0             13119622.0   
4                  487227.0              1239515.0   

   World Bank Forecast GDP(PPP)  CIA Forecast GDP(PPP)  Internet Users  \
0                     5326855.0              3875690.0     129800000.0   
1                     2273489.0              1742790.0      33950632.0   
2                     3837261.0              2989430.0     165300000.0   
3                    11874583.0              8443360.0     881250000.0   
4                     1225435.0               893310.0      40024673.0   

  UN Continental Region UN Statistical Subregion  Population 2022  \
0                Europe           Eastern Europe     1.447133e+08   
1              Americas         Northern America     3.845433e+07   
2              Americas            South America     2.153135e+08   
3                  Asia            Southern Asia     1.417173e+09   
4              Americas            South America     4.551032e+07   

   Population 2023  Population %Change  
0     1.444444e+08               -0.19  
1     3.878129e+07                0.85  
2     2.164224e+08                0.52  
3     1.428628e+09                0.81  
4     4.577388e+07                0.58

In [119]:
lnddf = data.sort_values('Land in km2', ascending=False)
lnddf

Country  Total in km2  Land in km2  Water in km2  Water %  \
0                   Russia    17098246.0   16378410.0    16378410.0      4.2   
1                   Canada     9984670.0    9093507.0     9093507.0      8.9   
2                   Brazil     8510346.0    8460415.0     8460415.0      0.6   
3                    India     3287263.0    2973190.0     2973190.0      9.6   
4                Argentina     2796427.0    2736690.0     2736690.0      1.6   
..                     ...           ...          ...           ...      ...   
137  Saint Kitts and Nevis         261.0        261.0         261.0      0.0   
138       Marshall Islands         181.0        181.0         181.0      0.0   
139          Liechtenstein         160.0        160.0         160.0      0.0   
140             San Marino          61.0         61.0          61.0      0.0   
141                 Tuvalu          26.0         26.0          26.0      0.0   

       HDI  %HDI Growth  IMF Forecast GDP(Nominal)  \
0    0.822         0.29                  1862470.0   
1    0.936         0.25                  2117805.0   
2    0.754         0.38                  2126809.0   
3    0.633         0.88                  3732224.0   
4    0.842         0.09                   621833.0   
..     ...          ...                        ...   
137  0.777         0.21                     1069.0   
138  0.639          NaN                      277.0   
139  0.935         0.22                        NaN   
140  0.853          NaN                     1998.0   
141  0.641         0.36                       63.0   

     World Bank Forecast GDP(Nominal)  UN Forecast GDP(Nominal)  \
0                           2240422.0                 1778782.0   
1                           2139840.0                 1988336.0   
2                           1920096.0                 1608981.0   
3                           3385090.0                 3201471.0   
4                            632770.0                  487227.0   
..                                ...                       ...   
137                             962.0                     861.0   
138                             280.0                     257.0   
139                            6608.0                       NaN   
140                               NaN                    2021.0   
141                              60.0                      60.0   

     IMF Forecast GDP(PPP)  World Bank Forecast GDP(PPP)  \
0                5056479.0                     5326855.0   
1                2378973.0                     2273489.0   
2                4101022.0                     3837261.0   
3               13119622.0                    11874583.0   
4                1239515.0                     1225435.0   
..                     ...                           ...   
137                 1760.0                        1406.0   
138                  263.0                         249.0   
139                    NaN                        4978.0   
140                 2872.0                        2022.0   
141                   63.0                          61.0   

     CIA Forecast GDP(PPP)  Internet Users UN Continental Region  \
0                3875690.0     129800000.0                Europe   
1                1742790.0      33950632.0              Americas   
2                2989430.0     165300000.0              Americas   
3                8443360.0     881250000.0                  Asia   
4                 893310.0      40024673.0              Americas   
..                     ...             ...                   ...   
137                 1240.0         44669.0              Americas   
138                  240.0         20560.0               Oceania   
139                    NaN         37201.0                Europe   
140                 2060.0         20100.0                Europe   
141                   50.0          5520.0               Oceania   

    UN Statistical Subregion  Population 2022  Population 2023  \


In [132]:
#filter countries with a total land area of 1m km2 or more
lnddf.loc[lnddf['Total in km2'] >= 1.e6]

Country  Total in km2  Land in km2  Water in km2  Water %    HDI  \
0         Russia    17098246.0   16378410.0    16378410.0      4.2  0.822   
1         Canada     9984670.0    9093507.0     9093507.0      8.9  0.936   
2         Brazil     8510346.0    8460415.0     8460415.0      0.6  0.754   
3          India     3287263.0    2973190.0     2973190.0      9.6  0.633   
4      Argentina     2796427.0    2736690.0     2736690.0      1.6  0.842   
5     Kazakhstan     2724910.0    2699700.0     2699700.0      0.9  0.811   
6        Algeria     2381741.0    2381741.0     2381741.0      0.0  0.745   
7         Mexico     1964375.0    1943945.0     1943945.0      1.0  0.758   
8      Indonesia     1910931.0    1811569.0     1811569.0      4.9  0.705   
10         Libya     1676198.0    1759540.0     1759540.0      0.0  0.718   
9          Sudan     1861484.0    1731671.0     1731671.0      7.0  0.508   
12      Mongolia     1564116.0    1553556.0     1553556.0      0.7  0.739   
11          Iran     1648195.0    1531595.0     1531595.0      7.1  0.774   
13          Peru     1285216.0    1279996.0     1279996.0      0.4  0.762   
15         Niger     1267000.0    1266700.0     1266700.0      0.0  0.400   
14          Chad     1284000.0    1259200.0     1259200.0      1.9  0.394   
16        Angola     1246700.0    1246700.0     1246700.0      0.0  0.586   
17          Mali     1240192.0    1220190.0     1220190.0      1.6  0.428   
18  South Africa     1221037.0    1214470.0     1214470.0      0.4  0.713   
20      Ethiopia     1104300.0    1096630.0     1096630.0      0.7  0.498   
21       Bolivia     1098581.0    1083301.0     1083301.0      1.4  0.692   
19      Colombia     1141748.0    1038700.0     1038700.0      8.8  0.752   
22    Mauritania     1030700.0    1025520.0     1025520.0      0.4  0.556   
23         Egypt     1002000.0     995450.0      995450.0      0.6  0.731   

    %HDI Growth  IMF Forecast GDP(Nominal)  World Bank Forecast GDP(Nominal)  \
0          0.29                  1862470.0                         2240422.0   
1          0.25                  2117805.0                         2139840.0   
2          0.38                  2126809.0                         1920096.0   
3          0.88                  3732224.0                         3385090.0   
4          0.09                   621833.0                          632770.0   
5          0.51                   259292.0                          220623.0   
6          0.30                   224107.0                          191913.0   
7          0.15                  1811468.0                         1414187.0   
8          0.55                  1417387.0                         1319100.0   
10         0.26                    40194.0                           45752.0   
9          0.40                    25569.0                           51662.0   
12         0.48                    18782.0                           16811.0   
11         0.35                   366438.0                          388544.0   
13         0.45                   264636.0                          242632.0   
15         1.54                    17073.0                           13970.0   
14         0.77                    12596.0                           12704.0   
16         1.27                    93796.0                          106714.0   
17         0.53                    21309.0                           18827.0   
18         0.50                   380906.0                          405870.0   
20         1.74                   155804.0                          126783.0   
21         0.40                    46796.0                           43069.0   
19         0.32                   363835.0                          343939.0   
22         0.79                    10357.0                           10375.0   
23         0.73                   398397.0                          476748.0   

    UN Forecast GDP(Nominal)  IMF Forecast GDP(PPP)  \
0                  1778782.

In [137]:
#only display large countries 
lnddf.loc[lnddf['Total in km2'] <= 1.e6, 'Country'] = 'Other Countries'
fig = px.pie(lnddf, names='Country', values = 'Total in km2', 
                title = 'Countries by land area, km2')
fig.show()
                

In [73]:
data.columns

Index(['Country', 'Total in km2', 'Land in km2', 'Water in km2', 'Water %',
       'HDI', '%HDI Growth', 'IMF Forecast GDP(Nominal)',
       'World Bank Forecast GDP(Nominal)', 'UN Forecast GDP(Nominal)',
       'IMF Forecast GDP(PPP)', 'World Bank Forecast GDP(PPP)',
       'CIA Forecast GDP(PPP)', 'Internet Users', 'UN Continental Region',
       'UN Statistical Subregion', 'Population 2022', 'Population 2023',
       'Population %Change'],
      dtype='object')

In [138]:
data.head()

Country  Total in km2  Land in km2  Water in km2  Water %    HDI  \
0     Russia    17098246.0   16378410.0    16378410.0      4.2  0.822   
1     Canada     9984670.0    9093507.0     9093507.0      8.9  0.936   
2     Brazil     8510346.0    8460415.0     8460415.0      0.6  0.754   
3      India     3287263.0    2973190.0     2973190.0      9.6  0.633   
4  Argentina     2796427.0    2736690.0     2736690.0      1.6  0.842   

   %HDI Growth  IMF Forecast GDP(Nominal)  World Bank Forecast GDP(Nominal)  \
0         0.29                  1862470.0                         2240422.0   
1         0.25                  2117805.0                         2139840.0   
2         0.38                  2126809.0                         1920096.0   
3         0.88                  3732224.0                         3385090.0   
4         0.09                   621833.0                          632770.0   

   UN Forecast GDP(Nominal)  IMF Forecast GDP(PPP)  \
0                 1778782.0              5056479.0   
1                 1988336.0              2378973.0   
2                 1608981.0              4101022.0   
3                 3201471.0             13119622.0   
4                  487227.0              1239515.0   

   World Bank Forecast GDP(PPP)  CIA Forecast GDP(PPP)  Internet Users  \
0                     5326855.0              3875690.0     129800000.0   
1                     2273489.0              1742790.0      33950632.0   
2                     3837261.0              2989430.0     165300000.0   
3                    11874583.0              8443360.0     881250000.0   
4                     1225435.0               893310.0      40024673.0   

  UN Continental Region UN Statistical Subregion  Population 2022  \
0                Europe           Eastern Europe     1.447133e+08   
1              Americas         Northern America     3.845433e+07   
2              Americas            South America     2.153135e+08   
3                  Asia            Southern Asia     1.417173e+09   
4              Americas            South America     4.551032e+07   

   Population 2023  Population %Change  
0     1.444444e+08               -0.19  
1     3.878129e+07                0.85  
2     2.164224e+08                0.52  
3     1.428628e+09                0.81  
4     4.577388e+07                0.58

In [142]:
hdi = data.sort_values('%HDI Growth', ascending = False)
hdi

Country  Total in km2  Land in km2  Water in km2  Water %  \
105             Haiti       27750.0      27560.0       27560.0      0.7   
20           Ethiopia     1104300.0    1096630.0     1096630.0      0.7   
54       Burkina Faso      274200.0     273800.0      273800.0      0.2   
66         Bangladesh      148460.0     134208.0      134208.0      9.6   
114          Eswatini       17363.0      17204.0       17204.0      0.9   
..                ...           ...          ...           ...      ...   
133          Barbados         431.0        431.0         431.0      0.0   
119            Brunei        5765.0       5265.0        5265.0      8.7   
115           Bahamas       13940.0      10010.0       10010.0     27.8   
138  Marshall Islands         181.0        181.0         181.0      0.0   
140        San Marino          61.0         61.0          61.0      0.0   

       HDI  %HDI Growth  IMF Forecast GDP(Nominal)  \
105  0.535         1.94                    25986.0   
20   0.498         1.74                   155804.0   
54   0.449         1.72                    20785.0   
66   0.661         1.64                   446349.0   
114  0.597         1.57                     4648.0   
..     ...          ...                        ...   
133  0.790         0.02                     6220.0   
119  0.829         0.01                    15153.0   
115  0.812         0.00                    13876.0   
138  0.639          NaN                      277.0   
140  0.853          NaN                     1998.0   

     World Bank Forecast GDP(Nominal)  UN Forecast GDP(Nominal)  \
105                           20254.0                   19044.0   
20                           126783.0                   99269.0   
54                            18885.0                   19738.0   
66                           460201.0                  414907.0   
114                            4854.0                    4732.0   
..                                ...                       ...   
133                            5638.0                    4844.0   
119                           16682.0                   14006.0   
115                           12897.0                   11209.0   
138                             280.0                     257.0   
140                               NaN                    2021.0   

     IMF Forecast GDP(PPP)  World Bank Forecast GDP(PPP)  \
105                38952.0                       36096.0   
20                393297.0                      346893.0   
54                 62788.0                       54978.0   
66               1476870.0                     1265940.0   
114                13797.0                       11508.0   
..                     ...                           ...   
133                 5436.0                        4263.0   
119                32000.0                       29415.0   
115                18146.0                       13538.0   
138                  263.0                         249.0   
140                 2872.0                        2022.0   

     CIA Forecast GDP(PPP)  Internet Users UN Continental Region  \
105                31620.0       1353986.0              Americas   
20                264050.0      19543075.0                Africa   
54                 45160.0       3047909.0                Africa   
66                793490.0     126210000.0                  Asia   
114                 9740.0        414278.0                Africa   
..                     ...             ...                   ...   
133                 3700.0        233604.0              Americas   
119                27230.0        406705.0                  Asia   
115                12100.0        336057.0              Americas   
138                  240.0         20560.0               Oceania   
140                 2060.0         20100.0                Europe   

    UN Statistical Subregion  Population 2022  Population 2023  \
105                Caribbean       11584996.0       117

In [149]:
fig = px.box(hdi, y= 'HDI', points='all')
fig.show()

From the boxplot above we can deduce the following:
-the median value is 0.72
-the data is not too varied and centres between 0.6 and 0.8
-the boxplot is fairly symmetrical and therefore evenly distributed around the median
These insights indicate that most countries are achieving close to the median HDI value. 

Let's compare GDP forcasts made by the IMF, both nominal and PPP

In [150]:
data.columns

Index(['Country', 'Total in km2', 'Land in km2', 'Water in km2', 'Water %',
       'HDI', '%HDI Growth', 'IMF Forecast GDP(Nominal)',
       'World Bank Forecast GDP(Nominal)', 'UN Forecast GDP(Nominal)',
       'IMF Forecast GDP(PPP)', 'World Bank Forecast GDP(PPP)',
       'CIA Forecast GDP(PPP)', 'Internet Users', 'UN Continental Region',
       'UN Statistical Subregion', 'Population 2022', 'Population 2023',
       'Population %Change'],
      dtype='object')

In [158]:
imfgdp = data[['Country', 'IMF Forecast GDP(Nominal)', 'IMF Forecast GDP(PPP)']].fillna(0)
imfgdp
              

Country  IMF Forecast GDP(Nominal)  IMF Forecast GDP(PPP)
0                   Russia                  1862470.0              5056479.0
1                   Canada                  2117805.0              2378973.0
2                   Brazil                  2126809.0              4101022.0
3                    India                  3732224.0             13119622.0
4                Argentina                   621833.0              1239515.0
..                     ...                        ...                    ...
137  Saint Kitts and Nevis                     1069.0                 1760.0
138       Marshall Islands                      277.0                  263.0
139          Liechtenstein                        0.0                    0.0
140             San Marino                     1998.0                 2872.0
141                 Tuvalu                       63.0                   63.0

[142 rows x 3 columns]

In [162]:
#let's sort the GDP values
nom = imfgdp.sort_values('IMF Forecast GDP(Nominal)', ascending=False)
nom

Country  IMF Forecast GDP(Nominal)  IMF Forecast GDP(PPP)
46          Germany                  4429838.0              5537992.0
45            Japan                  4230862.0              6495214.0
3             India                  3732224.0             13119622.0
57   United Kingdom                  3332059.0              3871790.0
51            Italy                  2186082.0              3193180.0
..              ...                        ...                    ...
76             Cuba                        0.0                    0.0
64            Syria                        0.0                    0.0
139   Liechtenstein                        0.0                    0.0
32      Afghanistan                        0.0                    0.0
71          Eritrea                        0.0                    0.0

[142 rows x 3 columns]

In [166]:
snom = nom.head(10)
snom

Country  IMF Forecast GDP(Nominal)  IMF Forecast GDP(PPP)
46         Germany                  4429838.0              5537992.0
45           Japan                  4230862.0              6495214.0
3            India                  3732224.0             13119622.0
57  United Kingdom                  3332059.0              3871790.0
51           Italy                  2186082.0              3193180.0
2           Brazil                  2126809.0              4101022.0
1           Canada                  2117805.0              2378973.0
0           Russia                  1862470.0              5056479.0
7           Mexico                  1811468.0              3277601.0
8        Indonesia                  1417387.0              4393370.0

In [169]:
fig = px.scatter(snom, x = 'IMF Forecast GDP(Nominal)', y ='IMF Forecast GDP(PPP)', color ='Country')
fig.show()

The scatter graph above shows the relationship between the nominal GDP figures and PPP GDP figures forecast for each country. PPP GDP adjusts for differences in price levels and provides a more accurate comparison of economic output and living standards. India has the highest GDP(PPP) forecast at over 12m, followed by Japan with a forecast of over 6m. In gerneal, this suggests that on average, goods and services in that country are less expensive compared to the prices in the reference country (typically, the US which serves as the base for PPP comparisons). Factors such as lower production costs and lower wages can attribute to this. Also, this means the local currency has a stronger purchasing power than indicated by the nominal exchange rate. 